### ISSUE TITLE
Not able to open info files other than Readme.info in WHDLoads

### ISSUE THREAD

**Issue by i30817** [libretro/libretro-uae#460]
This is kind of weird. I press 'z' on my keyboard (b button in retropad) to open the filechoose in a whdload game that has more than one info file (i used champions of krynn which has a subdirectory 'Docs' full of a ingame journal and a 'Docs.info' in the root, then i select one of the info files and press the 'ok button')

A screen appears too fast to read (normally) but i paused with RA 'F1' to see.

It says 
```
Couldn't load "ChampionsOfKrynn_v1.0_0986:ChampionsOfKrynn/Docs"
object is not of required type.
```

Then it fallbacks to reading the Readme.info.

**Comment by sonninnos** [comment]
Yes, the path is the culprit. Indeed launching `Docs.info` will not be possible, since it references the directory, and I don't think there is a way to filter those infos out, but I managed to fix launching infos under a directory.

Unfortunately that file requester forgets the initial pattern filter once the directory is changed, as in it will then show all files and not only infos..

And since this requires a change inside the WHDLoad helper image, it must be updated via deleting the directory/image after the core is updated.

You can test it by replacing `WHDLoad/S/WBSelect` file with this:
```
RequestFile #?.info >ENV:WBINFO
IF NOT "$WBINFO" EQ "" >NIL:
   List $WBINFO QUICK NOHEAD LFORMAT %E TO ENV:WBEXT
   IF "$WBEXT" EQ "info"
      List $WBINFO QUICK NOHEAD LFORMAT %F%M TO ENV:WBRUN
      WBRun "$WBRUN" SYNC >NIL:
   ELSE
      List $WBINFO QUICK NOHEAD LFORMAT %F%N TO ENV:WBRUN
      $WBRUN >NIL:
   ENDIF
ENDIF
UNSET WBINFO
```
It adds `%F` to the paths and silences warnings. Please confirm if there are more problems or not.

**Comment by i30817** [comment]
Seems to work.

BTW, i suspect, but do not know, the directory version might simply call /Docs/Journal.info if you select /Docs.info

**Comment by sonninnos** [comment]
Yes, falling back to readme when any launch fails is intentional.

I'll look into that directory info launching some more.

**Comment by sonninnos** [comment]
Okeydokey, new replacements:
[WHDLoad-update.zip](https://github.com/libretro/libretro-uae/files/7318202/WHDLoad-update.zip)

Changes being:
- Recursive info launching, as in Docs.info will start a new selector inside the directory, properly filtering out the unwanted files
- Summoning the selector once will return back to the selector after each launch
- Readme+MkCustom is only opened when launching Readme from the selector, or after quitting the game without the selector

Let me know if there are more annoyances or goofs. Too bad image files such as Map.info can't be opened  without some heavy dynamite, since proper Multiview program requires full Workbench to be running, and that would bloat the crap out of the helper image.. This current lightweight patent is simply a renamed text editor in its place.

**Comment by i30817** [comment]
It's cool. Any way to only allow slaves/executables (for hds)/infos to show if you select or go up a directory manually?

Ie: close all the gaps where the filter isn't applied.

**Comment by sonninnos** [comment]
Ah, that is not a WHDLoad game at all, and since it does not have identically named slave/info pair, the requester is forced automatically, as in it is not summoned, thus there is no remembering of the summoning. So if you hold the fire button down like usual, there will be no "double quit".

Same thing for example with `BattleIsle&DataDisks_v1.9_1991&0460&0667.lha`.

Add `SETENV WBSEL 1` to line 148 in `Startup-Sequence`, above `Execute S:WBSelect`.

**Comment by sonninnos** [comment]
I created MkCustom script for saving WHDLoad custom prefs, so that users are not forced to access the WHDLoad splash screen every time, and/or if the splash screen does not have custom pref checkboxes for cheats, intro skips etc.

It comes after Readme because the custom prefs are mentioned in the Readme.

**Comment by sonninnos** [comment]
I guess it could be more clear like `No custom WHDLoad options`, since if there are, it will show the current options.

**Comment by sonninnos** [comment]
How about:
![retroarch_2021_10_11_06_23_43_468](https://user-images.githubusercontent.com/45124675/136728947-97cf7038-ad75-4113-8e30-cc6c27897098.png)

![retroarch_2021_10_11_06_23_25_722](https://user-images.githubusercontent.com/45124675/136728956-48da8ebd-4ac7-45f6-b4e0-95d968e88078.png)

**Comment by sonninnos** [comment]
Okeydoke, I'll try to seal the deal tomorrow if nothing else comes up.
